### PR TITLE
moved getRecords & setRecords from InMemoryPipe to Pipe

### PR DESCRIPTION
**Pull request by chetan453**
Resolved issue #269

### CODE REVIEW

File: client/src/main/java/zingg/client/pipe/Pipe.java [hunk 1]
```diff
@@ -144,6 +147,13 @@ public void setId(int recId) {
 		this.id = recId;
 	}
 
+	public Dataset <Row> getRecords(){
```
Comment (sonalgoyal) [inline client/src/main/java/zingg/client/pipe/Pipe.java:150]: rename to getDataset, setDataset

File: api/scala/FebrlExample.scala [hunk 2]
```diff
@@ -10,61 +10,61 @@ val args = new Arguments();
 val fname = new FieldDefinition();
 fname.setFieldName("fname");
 fname.setDataType("\"string\"");
-fname.setMatchType(MatchType.FUZZY);
+fname.setMatchType(new ArrayList[MatchType](Arrays.asList(MatchType.FUZZY)));
```
Comment (sonalgoyal) [inline api/scala/FebrlExample.scala:13]: you should be able to do fname.setMatchType([MatchType.FUZZY]) - please check and apply for all the other match type changes below.

File: api/scala/FebrlExample.scala [hunk 3]
```diff
@@ -90,20 +90,18 @@ args.setLabelDataSampleSize(0.5f);
...
-val df = spark.read.option("header", "true").csv("examples/febrl/test.csv")
+val df = spark.read.format("csv").option("header",false).schema("id string, fname string, lname string, stNo string, add1 string, add2 string, city string, areacode string, state string, dob string, ssn  string").load("examples/febrl/test.csv")
 import zingg.client.pipe.InMemoryPipe;
 import java.util.HashMap
 
 val inputPipe = new InMemoryPipe(df);
 inputPipe.setProps(new HashMap[String, String]());
-val pipes = Array[zingg.client.pipe.Pipe](inputPipe);
-args.setData(pipes);
+args.setData(Array[zingg.client.pipe.Pipe](inputPipe));
```
Comment (sonalgoyal) [inline api/scala/FebrlExample.scala:99]: please revert and introduce pipes again as sometimes we may add more pipes to the input

File: api/scala/FebrlExample.scala [hunk 4]
```diff
@@ -90,20 +90,18 @@ args.setLabelDataSampleSize(0.5f);
...
 
 //setting outputpipe in 'args'
 val outputPipe = new InMemoryPipe();
 //outputPipe.setProps(new HashMap[String, String]());
-val pipes = Array[zingg.client.pipe.Pipe](outputPipe);
-args.setOutput(pipes);
+args.setOutput(Array[zingg.client.pipe.Pipe](outputPipe));
```
Comment (sonalgoyal) [inline api/scala/FebrlExample.scala:104]: please revert and add pipes again as we sometimes want to write to more than one pipe

**Review by sonalgoyal — CHANGES_REQUESTED** [review]

File: api/scala/FebrlExample.scala (hunk 2 above)
Comment (chetan453) [reply api/scala/FebrlExample.scala:13]: This is giving an error in my spark-shell.

**Pull request merged**